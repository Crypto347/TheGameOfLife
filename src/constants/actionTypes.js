export const CHOOSE_FROM_DROPDOWN = 'CHOOSE_FROM_DROPDOWN';
export const UPDATE_ARRAY_OF_CELLS = 'UPDATE_ARRAY_OF_CELLS';
export const UPDATE_ZOOM = 'UPDATE_ZOOM';
export const UPDATE_MAP_INTERACTION_CSS = 'UPDATE_MAP_INTERACTION_CSS';
export const START_ZOOMING = 'START_ZOOMING';
export const GAME_INITIALIZATION = 'GAME_INITIALIZATION';
export const MAKE_THE_CELL_ALIVE = 'MAKE_THE_CELL_ALIVE';
export const ON_NEXT_HANDLER = 'ON_NEXT_HANDLER';
export const ADD_OPTION = 'ADD_OPTION';
export const SAVE_PLAYING_AREA = 'SAVE_PLAYING_AREA';
export const UPDATE_PLAYING_AREA = 'UPDATE_PLAYING_AREA';
export const DELETE_OPTION = 'DELETE_OPTION';
