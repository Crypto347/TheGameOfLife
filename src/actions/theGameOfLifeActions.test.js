/**
* Libraries
*/

import configureStore from 'redux-mock-store';

/**
* Constants
*/

import * as Actions from '.';
import * as actionTypes from "../constants/actionTypes";

/**
* Tests
*/

const mockStore = configureStore();
const store = mockStore();

describe('movingBallsActions', () => {
    beforeEach(() => { // Runs before each test in the suite
      store.clearActions();
    });

    it('Dispatches the correct action and payload (moveCircleXCoordinate)', () => {
        const expectedActions = [
            {
                type: actionTypes.MOVE_CIRCLE_X_COORDINATE,
                dx: 467,
                id: 3
            }
        ];

        store.dispatch(Actions.moveCircleXCoordinate(467, 3));
        expect(store.getActions()).toEqual(expectedActions);
        expect(store.getActions()).toMatchSnapshot();
    });

    it('Dispatches the correct action and payload (moveCircleYCoordinate)', () => {
        const expectedActions = [
            {
                type: actionTypes.MOVE_CIRCLE_Y_COORDINATE,
                dy: 342,
                id: 2
            }
        ];

        store.dispatch(Actions.moveCircleYCoordinate(342, 2));
        expect(store.getActions()).toEqual(expectedActions);
        expect(store.getActions()).toMatchSnapshot();
    });

    it('Dispatches the correct action and payload (changeDirectionOfXMove)', () => {
        const expectedActions = [
            {
                type: actionTypes.CHANGE_DIRECTION_OF_X_MOVE,
                id: 7
            }
        ];

        store.dispatch(Actions.changeDirectionOfXMove(7));
        expect(store.getActions()).toEqual(expectedActions);
        expect(store.getActions()).toMatchSnapshot();
    });

    it('Dispatches the correct action and payload (changeDirectionOfYMove)', () => {
        const expectedActions = [
            {
                type: actionTypes.CHANGE_DIRECTION_OF_Y_MOVE,
                id: 3
            }
        ];

        store.dispatch(Actions.changeDirectionOfYMove(3));
        expect(store.getActions()).toEqual(expectedActions);
        expect(store.getActions()).toMatchSnapshot();
    });

    it('Dispatches the correct action and payload (fillCirclesArray)', () => {
        const expectedActions = [
            {
                type: actionTypes.FILL_CIRCLES_ARRAY
            }
        ];

        store.dispatch(Actions.fillCirclesArray());
        expect(store.getActions()).toEqual(expectedActions);
        expect(store.getActions()).toMatchSnapshot();
    });

    it('Dispatches the correct action and payload (updateCirclesArray)', () => {
        const expectedActions = [
            {
                type: actionTypes.UPDATE_CIRCLES_ARRAY,
                arr: [{a: 6},{b: 9}]
            }
        ];

        store.dispatch(Actions.updateCirclesArray([{a: 6},{b: 9}]));
        expect(store.getActions()).toEqual(expectedActions);
        expect(store.getActions()).toMatchSnapshot();
    });

    it('Dispatches the correct action and payload (getNumbersOfBalls)', () => {
        const expectedActions = [
            {
                type: actionTypes.GET_NUMBERS_OF_BALLS,
                num: 57
            }
        ];

        store.dispatch(Actions.getNumbersOfBalls(57));
        expect(store.getActions()).toEqual(expectedActions);
        expect(store.getActions()).toMatchSnapshot();
    });
});
