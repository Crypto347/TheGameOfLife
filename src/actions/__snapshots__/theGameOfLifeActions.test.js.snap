// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`theGameOfLifeReducer Dispatches the correct action and payload (startZooming) 1`] = `
Array [
  Object {
    "type": "START_ZOOMING",
    "zoom": 5,
  },
]
`;

exports[`theGameOfLifeReducer Dispatches the correct action and payload (updateArrayOfCells) 1`] = `
Array [
  Object {
    "array": Array [
      Object {
        "a": 2,
        "b": 7,
      },
    ],
    "type": "UPDATE_ARRAY_OF_CELLS",
  },
]
`;

exports[`theGameOfLifeReducer Dispatches the correct action and payload (updateMapInteractionCSS) 1`] = `
Array [
  Object {
    "type": "UPDATE_MAP_INTERACTION_CSS",
    "zoom": 2,
  },
]
`;

exports[`theGameOfLifeReducer Dispatches the correct action and payload (updateZoom) 1`] = `
Array [
  Object {
    "type": "UPDATE_ZOOM",
    "val": 3,
  },
]
`;
